packages/utils/ast-utils.test.js

```diff
@@ -210,4 +210,122 @@ const a = { plugs: [] }
 			expect(j(require).toSource()).toMatchSnapshot();
 		});
 	});
+
+	describe("safeTraverse", () => {
+		it("should safe traverse", () => {
+			const testObject = {
+				type: "NodeType"
+			};
+			const p = {
+				foo: {
+					bar: testObject
+				}
+			};
+			const require = utils.safeTraverse(p, ["foo", "bar"]);
+			expect(require).toEqual(testObject);
+		});
+
+		it("should safe traverse thrice", () => {
+			const type = {
+				type: "NodeType"
+			};
+			const p = {
+				parent: {
+					value: {
+						value: type
+					}
+				}
+			};
+			const traversedValue = utils.safeTraverse(p, ["parent", "value", "value"]);
+			expect(traversedValue).toEqual(type);
+		});
+	});
+
+	describe("safeTraverseAndGetType", () => {
+		it("should safe traverse and get type", () => {
+			const NODE_TYPE = "NodeType";
+			const p = {
+				value: {
+					value: {
+						type: NODE_TYPE
+					}
+				}
+			};
+			const typeValue = utils.safeTraverseAndGetType(p);
+			expect(typeValue).toEqual(NODE_TYPE);
+		});
+
+		it("should safe traverse and return false if not found", () => {
+			const NODE_TYPE = "NodeType";
+			const p = {
+				foo: {
+					bar: {
+						type: NODE_TYPE
+					}
+				}
+			};
+			const typeValue = utils.safeTraverseAndGetType(p);
```

Comment:
se the prev comment on a proper test case

Reply:
We are hardcoding `value.value` inside [`safeTraverseAndGetType()`](https://github.com/webpack/webpack-cli/pull/456/files#diff-0214ffef8038112039348669c8ba1baaR31), so the test has to be in that shape only.
Should we add an optional second arg for pathValues inputs and fall back to `values.values` when second arg is not present?